components/automate-cli/pkg/verifyserver/services/certificatevalidation/certificatevalidation_test.go

```diff
@@ -0,0 +1,520 @@
+t0WcQFnqRRtXOVDCZuBE4jgFJgLP1p7p7qI=
+-----END CERTIFICATE-----`
+
+	INVALID_BLOCK = `INVALID`
+
+	INVALID_KEY_FORMAT = `-----BEGIN RSA PRIVATE KEY-----
+MIICXAIBAAKBgQCjcGqTkOq0CR3rTx0ZSQSIdTrDrFAYl29611xN8aVgMQIWtDB/
+lD0W5TpKPuU9iaiG/sSn/VYt6EzN7Sr332jj7cyl2WrrHI6ujRswNy4HojMuqtfa
+b5FFDpRmCuvl35fge18OvoQTJELhhJ1EvJ5KUeZiuJ3u3YyMnxxXzLuKbQIDAQAB
+AoGAPrNDz7TKtaLBvaIuMaMXgBopHyQd3jFKbT/tg2Fu5kYm3PrnmCoQfZYXFKCo
+ZUFIS/G1FBVWWGpD/MQ9tbYZkKpwuH+t2rGndMnLXiTC296/s9uix7gsjnT4Naci
+5N6EN9pVUBwQmGrYUTHFc58ThtelSiPARX7LSU2ibtJSv8ECQQDWBRrrAYmbCUN7
+ra0DFT6SppaDtvvuKtb+mUeKbg0B8U4y4wCIK5GH8EyQSwUWcXnNBO05rlUPbifs
+DLv/u82lAkEAw39sTJ0KmJJyaChqvqAJ8guulKlgucQJ0Et9ppZyet9iVwNKX/aW
+9UlwGBMQdafQ36nd1QMEA8AbAw4D+hw/KQJBANJbHDUGQtk2hrSmZNoV5HXB9Uiq
+7v4N71k5ER8XwgM5yVGs2tX8dMM3RhnBEtQXXs9LW1uJZSOQcv7JGXNnhN0CQBZe
+nzrJAWxh3XtznHtBfsHWelyCYRIAj4rpCHCmaGUM6IjCVKFUawOYKp5mmAyObkUZ
+f8ue87emJLEdynC1CLkCQHduNjP1hemAGWrd6v8BHhE3kKtcK6KHsPvJR5dOfzbd
+HAqVePERhISfN6cwZt5p8B3/JUwSR8el66DF7Jm57BM=
+-----END RSA PRIVATE KEY-----`
+)
+
+func TestValidateCertificateService(t *testing.T) {
+	vc := certificatevalidation.NewValidateCertificateService(logger.NewTestLogger())
+	assert.NotNil(t, vc)
+}
+
+func TestCertificateValidation(t *testing.T) {
+	vc := certificatevalidation.NewValidateCertificateService(logger.NewTestLogger())
+	assert.NotNil(t, vc)
+	tests := []struct {
+		TestName        string
+		ReqBody         models.CertificateCheckRequest
+		ExpectedResBody models.CertificateCheckResponse
+	}{
+		{
+			TestName: "All Certificates are valid",
+			ReqBody: models.CertificateCheckRequest{
+				RootCertificate:  ROOT_CERTIFICATE,
+				PrivateKey:       PRIVATE_KEY,
+				NodeCertificate:  NODE_CERTIFICATE,
+				AdminPrivateKey:  ADMIN_PRIVATE_KEY,
+				AdminCertificate: ADMIN_CERTIFICATE,
+			},
+			ExpectedResBody: models.CertificateCheckResponse{
+				Passed: true,
+				Checks: []models.Checks{
+					{
+						Title:         constants.CERTIFICATE_EXPIRY_TITLE,
+						Passed:        true,
+						SuccessMsg:    constants.CERTIFICATE_EXPIRY_SUCCESS_MESSAGE,
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+					{
+						Title:         constants.CERTIFICATE_FORMAT_TITLE,
+						Passed:        true,
+						SuccessMsg:    constants.CERTIFICATE_FORMAT_SUCCESS_MESSAGE,
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+					{
+						Title:         constants.KEY_FORMAT_TITLE,
+						Passed:        true,
+						SuccessMsg:    constants.KEY_FORMAT_SUCCESS_MESSAGE,
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+					{
+						Title:         constants.CERTIFICATE_ALGORITHM_TITLE,
+						Passed:        true,
+						SuccessMsg:    constants.CERTIFICATE_ALGORITHM_SUCCESS_MESSAGE,
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+				},
+			},
+		},
+		{
+			TestName: "Root Certificate is about to expire, not in x509 v3 format, uses incorrect Hash Algo and Node Certificate is expired.",
```

Comment:
Can we have similar test for node and admin certificate?

Reply:
done